src/migrations/tasks/20200305134738_swu_opportunities.ts

```diff
@@ -0,0 +1,170 @@
+    ALTER TABLE "swuOpportunityStatuses"
+    ADD CONSTRAINT "eitherEventOrStatus" check(
+      ("event" IS NOT NULL AND "status" IS NULL)
+      OR
+      ("event" IS NULL AND "status" IS NOT NULL)
+    )
+  `);
+
+  logger.info('Created swuOpportunityStatuses table.');
+
+  await connection.schema.createTable('swuOpportunityAttachments', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.uuid('opportunityVersion').references('id').inTable('swuOpportunityVersions').notNullable().onDelete('CASCADE');
+    table.uuid('file').references('id').inTable('files').notNullable();
+  });
+
+  logger.info('Created swuOpportunityAttachments table.');
+
+  await connection.schema.createTable('swuOpportunityAddenda', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.uuid('opportunity').references('id').inTable('swuOpportunities').notNullable().onDelete('CASCADE');
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.string('description').notNullable();
+  });
+  logger.info('Created swuOpportunityAddenda table.');
+
+  await connection.schema.createTable('swuOpportunityPhases', table => {
+    table.uuid('id').primary().unique().notNullable();
+    table.uuid('opportunityVersion').references('id').inTable('swuOpportunityVersions').notNullable().onDelete('CASCADE');
+    table.enu('phase', Object.values(SWUOpportunityPhaseType)).notNullable();
+    table.timestamp('startDate').notNullable();
+    table.timestamp('completionDate').notNullable();
+    table.integer('maxBudget').notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.timestamp('updatedAt').notNullable();
+    table.uuid('updatedBy').references('id').inTable('users').notNullable();
+
+    table.unique(['opportunityVersion', 'phase']);
+  });
+  logger.info('Created swuOpportunityPhases table.');
+
+  await connection.schema.createTable('swuPhaseCapabilities', table => {
+    table.uuid('phase').references('id').inTable('swuOpportunityPhases').notNullable().onDelete('CASCADE');
+    table.enum('capability', SKILLS).notNullable();
+    table.boolean('fullTime').defaultTo(false).notNullable();
+    table.timestamp('createdAt').notNullable();
+    table.uuid('createdBy').references('id').inTable('users').notNullable();
+    table.timestamp('updatedAt').notNullable();
+    table.uuid('updatedBy').references('id').inTable('users').notNullable();
+    table.primary(['phase', 'capability']);
+  });
+  logger.info('Created swuPhaseCapabilities table.');
+
+  await connection.schema.createTable('swuTeamQuestions', table => {
+    table.uuid('opportunityVersion').references('id').inTable('swuOpportunityVersions').notNullable().onDelete('CASCADE');
+    table.string('question', 1000).notNullable();
+    table.string('guideline', 1000).notNullable();
```

Comment:
Are the `1000`s char limits? Don't add these in the DM, this should be validated using validation functions, easier to change as needed in the future.

Reply:
They are char limits, but knex defaults to 255 if you don't specify.  We may need more than that for those items.

Reply:
In past migrations, we create quite a few string columns without specifying the char limit — these should probably be changed too? Also, does postgres support a "text" column type without the char limit?

Reply:
Don't make the change yet — let's figure out a solution first

Reply:
We should be using text.  From what I'm seeing in the database, Knex with Postgres will still only allow 255 max despite specifying the length because it uses varchar, which has a hard limit of 255 characters.  Text does not have a limit.

I can do a migration to update our larger string fields to use text instead.  Thanks for catching this.

Reply:
Will issue new migration in a new PR as discussed.